In [1]:
data <- as.numeric(as.matrix(read.csv("data.txt", header=FALSE)))

In [2]:
n = length(data)
x = data
yk <- function(k) {
    return(sqrt(k*(n-k)/n^2)*(mean(x[1:k])-mean(x[(k+1):n])))
}

In [3]:
y = NULL
for (i in 1:n-1){
    y = c(y,abs(yk(i)))
}
max(y)>1.36/sqrt(n)
round(max(y),5)
which(y == max(y))

[1] TRUE

[1] 0.09103

[1] 503

The test shows that the dataset is not statistically homogeneous when assuming that the confidence level is 95 %.

$Y_n^{max} = 0.09103$

The estimated change-point is 503-rd point.